dotCMS/src/main/java/com/dotmarketing/portlets/languagesmanager/business/LanguageAPIImpl.java

```diff
@@ -167,88 +196,123 @@ public void saveLanguageKeys(Language lang, Map<String, String> generalKeys, Map
-            Logger.error( this, e.getMessage(), e );
-        }
-
-        //If we didn't find a value for the given language, lets try with the default one
-        if ( value == null ) {
-            try {
-                value = LanguageUtil.get( user, key );//Searching this key in the default language
-            } catch ( LanguageException e ) {
-                Logger.error( this, e.getMessage(), e );
-            }
-
-        }
-
-        return value;
-    }
+		return user;
+	}
 
-	public boolean getBooleanKey(Language lang, String key) {
+	@Override
+	public boolean getBooleanKey(final Language lang, final String key) {
 		return Boolean.parseBoolean(getStringKey(lang, key));
 	}
 
-	public boolean getBooleanKey(Language lang, String key, boolean defaultVal) {
-		if(getStringKey(lang, key) != null)
-			return Boolean.parseBoolean(getStringKey(lang, key));
-		return defaultVal;
+	@Override
+	public boolean getBooleanKey(final Language lang, final String key, final boolean defaultVal) {
+		return (getStringKey(lang, key) != null)?
+			Boolean.parseBoolean(getStringKey(lang, key)):defaultVal;
 	}
 
-	public float getFloatKey(Language lang, String key) {
+	@Override
+	public float getFloatKey(final Language lang, final String key) {
 		return Float.parseFloat(getStringKey(lang, key));
 	}
 
-	public float getFloatKey(Language lang, String key, float defaultVal) {
-		if(getStringKey(lang, key) != null)
-			return Float.parseFloat(getStringKey(lang, key));
-		return defaultVal;
+	@Override
+	public float getFloatKey(final Language lang, final String key, final float defaultVal) {
+		return (getStringKey(lang, key) != null)?
+			Float.parseFloat(getStringKey(lang, key)):defaultVal;
 	}
 
-	public int getIntKey(Language lang, String key) {
+	@Override
+	public int getIntKey(final Language lang, final String key) {
 		return Integer.parseInt(getStringKey(lang, key));
 	}
 
-	public int getIntKey(Language lang, String key, int defaultVal) {
-		if(getStringKey(lang, key) != null)
-			return Integer.parseInt(getStringKey(lang, key));
-		return defaultVal;
+	@Override
+	public int getIntKey(final Language lang, final String key, final int defaultVal) {
+		return (getStringKey(lang, key) != null)?
+                Integer.parseInt(getStringKey(lang, key)):defaultVal;
 	}
 
+	@Override
 	public void clearCache(){
 		CacheLocator.getLanguageCache().clearCache();
 	}
+
+    @Override
+    public Language getFallbackLanguage(final String languageCode) {
+        return this.factory.getFallbackLanguage(languageCode);
+    }
+
+    /**
+     * Utility method used to get an instance of the {@link LanguageVariableAPI}. This is used to
+     * avoid Stackoverflow exceptions when starting up the system.
+     * 
+     * @return An instance of the {@link LanguageVariableAPI}.
+     */
+    private LanguageVariableAPI getLanguageVariableAPI() {
+
+    	if (null == this.languageVariableAPI) {
+
+        	synchronized (this) {
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Remove this dangerous instance of double-checked locking. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2168)